### ISSUE TITLE
Immediate non-continous flush

### ISSUE THREAD

**Issue by maswan** [dCache/dcache#3659]
For some HSM providers (like ENDIT), batching into flush rounds falls much more natural in the HSM provider, and in order to not have double-waiting it would be nice if there was a mode where dCache would just issue stores as it had precious files, and keep doing that for new files even if there was outstanding stores instead of the current batching behaviour where one set of stores must finish before any new are created.

I know there had been talk about this previously, but I could find neither support for it nor an issue about it. If that's a mistake, please correct me and close this. :)

**Comment by paulmillar** [comment]
Yes, you're quite right.  We've talked about this internally, but AFAIK, no progress was made.

We would benefit from this, too, here at DESY as we have a custom 'tape' script that packs small files together.

**Comment by kofemann** [comment]
Did you try to configure dCache to flush right away into 'open-ended' flush queue:

```
hsm create osm osm script  -command=/usr/share/dcache/lib/osm-hsmcp.py -c:gets=10000 -c:puts=10000 -command=...
queue define class osm * -pending=1 -total=1 -expire=1 -open=true
``` 

This tells dCache to flush files even if it's only one (1) (pending), even if it's only one (1) byte (total) and even if it's only one (1) second old (expire). You can set '-c:puts' and '-c:gets' to a big number to simulate unlimited queue. The '-open=true' will take care that new request just added without waiting until all **in progress** requests to finish.

**Comment by maswan** [comment]
Thank you, that was what I was missing! It does in fact work fine with this setting, with one comment:

I had to define the class for the specific storage class, "*" didn't work.

From my point of view this is solved then.

**Comment by XMol** [comment]
Hi @kofemann,

did you by any chance follow up on your "I will fix that" promise made in RT \#10213? 😉 

The match-all `*` does not respect the `-open` flag, is what you said back then.

Ciao,  
Xavier.

**Comment by XMol** [comment]
This issue was addressed by 90206858a713d794b6920395e6e73aa7f9819ed2 and has been merged into the mainline releases already (6.2.36, 7.0.25, 7.1.16 and 7.2.6).

This issue then may be closed (by someone who has the rights to do so). 🙂
